server/src/user/user-service.ts

```diff
@@ -59,7 +59,13 @@ export class UserService {
 
         user.emailConfirmed = false;
         user.enableEmailNotifications = false;
-
+        
         await this.userRepository.persistAndFlush(user);
     }
+
+    public async deleteUserAccount(userId: string){
+        const user = await this.userRepository.findOneOrFail(userId);
+
+        await this.userRepository.nativeDelete(user);
```

Comment:
```suggestion
        await this.userRepository.nativeDelete(userId);
```
No need to fetch the whole entity for that operation ;)